Caltech-IPAC/firefly
# DM-7225 File Upload Function for FITS, VOTable, IPAC, CSV, TSV files.

The implementation includes: 
- a generic file upload function for files of FITS, VOTable, IPAC, CSV, TSV formats. 
  . for FITS format file, the fits could consist of one or more HDU units. 
    for the FITS with extensions, the extension could be either image or table type, or both. 
  . for VOTable file, the file could contain data of 'TABLEDATA', 'BINARY', 'BINARY2', or 'FITS' format. 

test: 
- localhost:8080/firefly
- select 'File Upload' tab
- upload file by clicking 'Choose File' 
   (there are some samples put under lsst/firefly_test_data/FileUpload-samples/CSV, TSV, IPAC, FITS, VOTABLE)
 - for votable and fits files, a summary table which reports the 'units' of the uploaded file is shown below the 'Choose File' button. 
    By highlighting the table row, the header of FITS HDU or the information of the table in VOTable is shown at the right of the summary table. 
    By selecting the table row(s) and clicking 'Search' button, the image or the table content will be further shown in the tri-view panel. 
- for CSV, TSV, IPAC files. the table content is rendered on the tri-view by clicking 'Search'
